src/modules/claims/claims.service.ts

```diff
@@ -575,25 +575,26 @@ export class ClaimsService {
 
   // TODO: create docs annotations
   async getClaimId({ claimData }: { claimData: ClaimData }) {
-    const { service = [] } = await this._didRegistry.getDidDocument();
-    const { id, claimTypeVersion } =
-      service.find(
-        ({ profile, claimType, claimTypeVersion }) =>
-          Boolean(profile) ||
-          (claimType === claimData.claimType &&
-            claimTypeVersion === claimData.claimTypeVersion)
-      ) || {};
+    const services = await this._didRegistry.getServices();
+    const service = services.find(
+      ({ profile, claimType, claimTypeVersion }) =>
+        Boolean(profile) ||
+        (claimType === claimData.claimType &&
+          claimTypeVersion === claimData.claimTypeVersion)
+    );
+
+    if (!service) return v4();
 
     if (claimData.profile && id) {
-      return id;
+      return service.id;
     }
 
     if (
       claimData.claimType &&
-      id &&
-      claimData.claimTypeVersion === claimTypeVersion
+      service.id &&
+      claimData.claimTypeVersion === service.claimTypeVersion
```

Comment:
May be we remove this now? From the way the service was found implies that this conditions are satisfied

Reply:
I don't want to change the logic here, maybe just little refactors (that's all). So create a ticket in jira pls